sklearn/cluster/k_means_.py

```diff
@@ -552,17 +553,16 @@ def _labels_inertia_precompute_dense(X, x_squared_norms, centers, distances):
 
     """
     n_samples = X.shape[0]
-    k = centers.shape[0]
-    all_distances = euclidean_distances(centers, X, x_squared_norms,
-                                        squared=True)
-    labels = np.empty(n_samples, dtype=np.int32)
-    labels.fill(-1)
-    mindist = np.empty(n_samples)
-    mindist.fill(np.infty)
-    for center_id in range(k):
-        dist = all_distances[center_id]
-        labels[dist < mindist] = center_id
-        mindist = np.minimum(dist, mindist)
+
+    # Breakup nearest neighbor distance computation into batches to prevent
+    # memory blowup
+    metric_kwargs = dict(squared=True)
+    # Should use check_inputs=False when speedup_kmpp is merged
```

Comment:
I would rather not mention the other PR here, if you want to do so, please use the PR number. There's no way for someone else to look up the issue based on your branch name (as far as I know)